packages/medusa/src/api/routes/store/customers/claim-orders.ts

```diff
@@ -0,0 +1,134 @@
+import { SamlTokenUserAccountAuthMutationVariables } from "@linear/sdk/dist/_generated_documents"
+import { IsNotEmpty, IsString } from "class-validator"
+import { MedusaError } from "medusa-core-utils"
+import { EntityManager } from "typeorm"
+import { Order } from "../../../../models"
+import {
+  CustomerService,
+  EventBusService,
+  OrderService,
+} from "../../../../services"
+import TokenService from "../../../../services/token"
+
+/**
+ * @oas [post] /customers/claim-orders
+ * operationId: "PostCustomersCustomerOrderClaim"
+ * summary: "Claim orders for signed in account"
+ * description: "Sends an email to emails registered to orders provided with link to transfer order ownership"
+ * requestBody:
+ *   content:
+ *     application/json:
+ *       schema:
+ *         required:
+ *           - order_ids
+ *         properties:
+ *           order_ids:
+ *             description: "The ids of the orders to claim"
+ *             type: array
+ *             items:
+ *              type: string
+ * x-codeSamples:
+ *   - lang: JavaScript
+ *     label: JS Client
+ *     source: |
+ *       import Medusa from "@medusajs/medusa-js"
+ *       const medusa = new Medusa({ baseUrl: MEDUSA_BACKEND_URL, maxRetries: 3 })
+ *       // must be previously logged in or use api token
+ *       medusa.customers.claimOrders({
+ *         order_ids,
+ *       })
+ *       .then(() => {
+ *         // successful
+ *       })
+ *       .catch(() => {
+ *         // an error occurred
+ *       });
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request POST 'https://medusa-url.com/store/customers/claim-orders' \
+ *       --header 'Content-Type: application/json' \
+ *       --data-raw '{
+ *           "order_ids": ["id"],
+ *       }'
+ * security:
+ *   - api_token: []
+ *   - cookie_auth: []
+ * tags:
+ *   - Invite
+ * responses:
+ *   200:
+ *     description: OK
+ *   "400":
+ *     $ref: "#/components/responses/400_error"
+ *   "401":
+ *     $ref: "#/components/responses/unauthorized"
+ *   "404":
+ *     $ref: "#/components/responses/not_found_error"
+ *   "409":
+ *     $ref: "#/components/responses/invalid_state_error"
+ *   "422":
+ *     $ref: "#/components/responses/invalid_request_error"
+ *   "500":
+ *     $ref: "#/components/responses/500_error"
+ */
+export default async (req, res) => {
+  const { order_ids } = req.validatedBody
```

Comment:
thought: This approach requires the customer to know all of their order ids, which I think is an unreasonable expectation. Would it make sense to use display ids instead, as this is typically what you are presented with in your confirmation email?

Reply:
Phil and I talked about this - but since the display ids are simple numbers we would expose ourselves to enumeration attacks. Someone could just try to claim orders on every number from 1-1000 which would: 

1) send out a lot of emails and
2) potentially risk that someone accidentally hands over their data

Instead, we should surface the order id somewhere in customer communication to make it possible to do things like this. Wdyt?

Reply:
I see. Makes sense!